lib/media/ver10/get_stream_uri.ex

```diff
@@ -2,10 +2,14 @@ defmodule Onvif.Media.Ver10.GetStreamUri do
   import SweetXml
   import XmlBuilder
 
+  alias Onvif.Device
+
   def soap_action, do: "http://www.onvif.org/ver10/media/wsdl/GetStreamUri"
 
-  def request(uri, auth \\ :xml_auth, args),
-    do: Onvif.Media.Ver10.Media.request(uri, args, auth, __MODULE__)
+  @spec request(Device.t(), :basic_auth | :digest_auth | :no_auth | :xml_auth, list) ::
+          {:ok, any} | {:error, map()}
+  def request(device, auth \\ :xml_auth, args),
```

Comment:
The auth can also be taken from the device struct. I will address this in a follow up PR.

Reply:
Just remember to override the GetSystemDateAndTime call to `:no_auth` regardless of what is in Device